core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,251 @@
+@Description("A Hydrator Action plugin to efficiently export data from Oracle to HDFS or local file system." +
+  "The plugin uses Oracle's command line tools to export data." +
+  "The data exported from this tool can then be used in Hydrator pipelines.")
+public class OracleExportAction extends Action {
+  enum SeparatorFormat {
+    csv, tsv, psv
+  };
+  private final OracleExportActionConfig config;
+
+  public OracleExportAction(OracleExportActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    String oracleExportCommand = buildOracleExportCommand();
+    Connection connection = new Connection(config.oracleServerHostname, config.oracleServerPort);
+    try {
+      connection.connect();
+      boolean isAuthenticated = connection.authenticateWithPassword(config.oracleServerUsername,
+                                                                    config.oracleServerPassword);
+      if (!isAuthenticated) {
+
+        throw new IllegalStateException(String.format("SSH authentication error when connecting to %s@%s on port %d",
+                                            config.oracleServerUsername, config.oracleServerHostname,
+                                            config.oracleServerPort));
+      }
+      Session session = connection.openSession();
+      session.execCommand(oracleExportCommand);
+      try (InputStream stdout = new StreamGobbler(session.getStdout());
+           BufferedReader outBuffer = new BufferedReader(new InputStreamReader(stdout, Charsets.UTF_8))) {
+        Integer exitCode = session.getExitStatus();
+        if (exitCode != null && exitCode != 0) {
+          throw new IOException(String.format("Error running command %s on hostname %s; exit code: %d",
+                                              oracleExportCommand, config.oracleServerHostname, exitCode));
+        }
+        String out = CharStreams.toString(outBuffer);
+        //SQLPLUS command errors are not fetched from session.getStderr().
+        //Errors and output received after executing the command in SQLPlus prompt are the one that
+        //are printed on the SQL prompt
+        if (out.contains("ERROR at line")) {
+          throw new IOException(String.format("Error executing sqlplus query %s on hostname %s; error message: %s",
+                                              config.queryToExecute, config.oracleServerHostname, out));
+        }
+        Path file = new Path(config.outputPath);
+        FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+        try (FSDataOutputStream outStream = fs.create(file);
+          BufferedWriter br = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"))) {
+          br.write(out.replaceAll("(?m)^[\\s&&[^\\n]]+|[\\s+&&[^\\n]]+$", "")); //Removes multiline trailing spaces
```

Comment:
is this because you have set the linesize to a large number, so there are trailing spaces? If so, you should set ``trimspool`` and ``trimout`` in the sqlplus command so you don't need this complicated regex. Also, please test with ``trimspool`` and ``trimout`` to make sure it works.